packages/server-core/src/media/storageprovider/ipfs.storage.ts

```diff
@@ -0,0 +1,399 @@
+          status.push(true)
+        }
+      } catch {
+        status.push(false)
+      }
+    }
+
+    return status
+  }
+  createInvalidation = async (): Promise<any> => Promise.resolve()
+  async listFolderContent(folderName: string, recursive?: boolean): Promise<FileContentType[]> {
+    const filePath = path.join(this._pathPrefix, folderName)
+
+    const results: FileContentType[] = []
+
+    if (recursive) {
+      await this._parseMFSDirectoryAsType(filePath, results)
+    } else {
+      for await (const file of this._client.files.ls(filePath)) {
+        const signedUrl = await this.getSignedUrl(file.cid.toString(), 3600, null)
+
+        const res: FileContentType = {
+          key: file.cid.toString(),
+          name: file.name,
+          type: file.type,
+          url: signedUrl.url,
+          size: this._formatBytes(file.size)
+        }
+
+        results.push(res)
+      }
+    }
+
+    return results
+  }
+  async moveObject(oldName: string, newName: string, oldPath: string, newPath: string, isCopy?: boolean): Promise<any> {
+    const oldFilePath = path.join(this._pathPrefix, oldPath, oldName)
+    const newFilePath = path.join(this._pathPrefix, newPath, newName)
+
+    try {
+      if (isCopy) {
+        await this._client.files.cp(oldFilePath, newFilePath, { parents: true })
+      } else {
+        await this._client.files.mv(oldFilePath, newFilePath, { parents: true })
+      }
+    } catch (err) {
+      return false
+    }
+
+    return true
+  }
+
+  async initialize(podName: string): Promise<void> {
+    if (config.kubernetes.enabled) {
+      const kc = new k8s.KubeConfig()
+      kc.loadFromDefault()
+
+      const forward = new k8s.PortForward(kc)
+
+      this.cacheDomain = await this._forwardIPFS(podName, forward, 8080)
+      this._apiDomain = await this._forwardIPFS(podName, forward, 5001)
+      this._client = create({ url: `http://${this._apiDomain}` })
+      this._blobStore = new IPFSBlobStore(this._client)
+    }
+  }
+
+  async getIPFSPod(): Promise<string> {
+    if (config.kubernetes.enabled) {
+      const kc = new k8s.KubeConfig()
+      kc.loadFromDefault()
+
+      const k8DefaultClient = kc.makeApiClient(k8s.CoreV1Api)
+
+      const appName = `${config.server.releaseName}-ipfs`
+      const podsResult = await k8DefaultClient.listNamespacedPod(
+        'default',
+        undefined,
+        undefined,
+        undefined,
+        undefined,
+        `app.kubernetes.io/instance==${appName}`
+      )
+
+      if (podsResult.body.items.length > 0) {
+        return podsResult.body.items[0].metadata!.name!
+      }
+    }
+
+    return ''
+  }
+
+  async _forwardIPFS(podName: string, forward: k8s.PortForward, forwardPort: number): Promise<string> {
+    return new Promise((resolve) => {
+      const server = net.createServer((socket) => {
+        forward.portForward('default', podName, [forwardPort], socket, socket, socket)
+      })
+      server.listen(0, '127.0.0.1', () => {
```

Comment:
Should this IP be hardcoded here?

Reply:
this is how it communicates internally with k8s